packages/light.js/src/rpc/other/makeContract.ts

```diff
@@ -46,15 +49,15 @@ const getContract = memoizee(
  * function resolves.
  */
 export const makeContract = memoizee(
-  (address: Address, abiJson: any[]) => {
-    // use types from @parity/abi
-    const abi = new Abi(abiJson);
+  (address: Address, abiJson: any[], options: { provider?: any } = {}) => {
+    const { provider } = options;
+    const abi = new Abi(abiJson); // use types from @parity/abi
```

Comment:
memoization needs to be done based on address and api (same as above)

Reply:
so it's a bit more tricky here
we could have `getContract` require an api/provider to be passed as parameter; put `const api = provider ? createApiFromProvider(provider) : getApi();` here (in makeContract) and memoize based on `address` and `api`